src/imageLoader/wadouri/loadImage.js

```diff
@@ -37,20 +37,34 @@ function loadImageFromPromise (dataSetPromise, imageId, frame, sharedCacheKey, o
   const deferred = $.Deferred();
 
   dataSetPromise.then(function (dataSet/* , xhr*/) {
-    const pixelData = getPixelData(dataSet, frame);
-    const transferSyntax = dataSet.string('x00020010');
-    const loadEnd = new Date().getTime();
-    const imagePromise = createImage(imageId, pixelData, transferSyntax, options);
-
-    imagePromise.then(function (image) {
-      image.data = dataSet;
-      const end = new Date().getTime();
-
-      image.loadTimeInMS = loadEnd - start;
-      image.totalTimeInMS = end - start;
-      addDecache(image);
-      deferred.resolve(image);
-    });
+    try {
```

Comment:
This is minor, but the try/catch block seems unnecessarily large here. What were you seeing errors from? Maybe it makes more sense to just put it around the getPixelData and createImage calls?

I'm also a bit unsure about having the deferred rejection have two formats (object with error + dataset, or just error object). That said, it definitely makes sense to include the dataset when rejecting it when we have that information. This might be a stupid idea, but maybe we should just attach the dataset to the instance of the Error object? (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error)

Reply:
I pushed a new version that restricts the try/catch block and rejects `{ error }` instead of `error`, to have an unique format.